src/tests/illumidesk/conftest.py

```diff
@@ -296,16 +307,350 @@ def test_quart_client(monkeypatch, tmp_path):
     return app.test_client()
 
 
-@pytest.fixture
-def http_async_httpclient_with_simple_response(request):
+@pytest.fixture(scope='function')
+def make_mock_request_handler() -> RequestHandler:
+    """
+    Sourced from https://github.com/jupyterhub/oauthenticator/blob/master/oauthenticator/tests/mocks.py
+    """
+
+    def _make_mock_request_handler(
+        handler: RequestHandler, uri: str = 'https://hub.example.com', method: str = 'GET', **settings: dict
+    ) -> RequestHandler:
+        """Instantiate a Handler in a mock application"""
+        application = Application(
+            hub=Mock(base_url='/hub/', server=Mock(base_url='/hub/'),),
+            cookie_secret=os.urandom(32),
+            db=Mock(rollback=Mock(return_value=None)),
+            **settings,
+        )
+        request = HTTPServerRequest(method=method, uri=uri, connection=Mock(),)
+        handler = RequestHandler(application=application, request=request,)
+        handler._transforms = []
+        return handler
+
+    return _make_mock_request_handler
+
+
+@pytest.fixture(scope='function')
+def make_http_response() -> HTTPResponse:
+    async def _make_http_response(
+        handler: RequestHandler,
+        code: int = 200,
+        reason: str = 'OK',
+        headers: HTTPHeaders = HTTPHeaders({'content-type': 'application/json'}),
+        effective_url: str = 'http://hub.example.com/',
+        body: Dict[str, str] = {'foo': 'bar'},
+    ) -> HTTPResponse:
+        """
+        Creates an HTTPResponse object from a given request. The buffer key is used to
```

Comment:
I would agree with you that perhaps this is overkill docstring-wise considering its a fixture, perhaps something more summarized would do.